core/src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -151,20 +249,24 @@ public void onSourceRemoved(SourceRemovedEvent event) {
         }
     }
 
-    public synchronized void addStep(int index, Step step) {
+    public void addStep(int index, Step step) {
```

Comment:
So, this method is a wrapper around a wrapper around a wrapper around a wrapper around `lock.lock()` and `lock.unlock()`.  I personally found this difficult to follow, and it seems to have significantly increased the line count in this file.  I wonder if maybe it might be a good idea to use less layers of abstraction here, and just call `lock()` and `unlock()` directly.